classes/shop/Shop.php

```diff
@@ -396,7 +396,7 @@ public static function initialize()
             Configuration::getMultiShopValues('PS_MEDIA_SERVER_3')
         );
 
-        if ((!$id_shop && defined('_PS_ADMIN_DIR_')) || Tools::isPHPCLI() || in_array($http_host, $all_media)) {
+        if ((!$id_shop && defined('_PS_ADMIN_DIR_')) || ('all' === $id_shop && defined('_PS_API_IN_USE_') && _PS_API_IN_USE_) || Tools::isPHPCLI() || in_array($http_host, $all_media)) {
```

Comment:
I suggest to use vars to improve the code readability 
Something like :

```php
$isAllShop = 'all' === $id_shop;
$isApiInUse = defined('_PS_API_IN_USE_') && _PS_API_IN_USE_;
```

Reply:
I updated the commit with your suggestions.